clap_complete/tests/testsuite/dynamic.rs

```diff
@@ -45,6 +45,52 @@ fn suggest_hidden_long_flags() {
     assert_data_eq!(complete!(cmd, "--hello-m"), snapbox::str!["--hello-moon"]);
 }
 
+#[test]
+fn suggest_hidden_subcommand_and_aliases() {
```

Comment:
This commit seems unrelated to the title of the PR.  Let's split this out into a separate PR and focus on getting the existing functionality merged

(and please do the two commit thing with tests)